src/redux/books/books.js

```diff
@@ -0,0 +1,35 @@
+const ADD = 'bookstore/books/ADD';
+const REMOVE = 'bookstore/books/REMOVE';
+const INITIAL_STATE = [];
+
+export function addBook(book) {
+  return {
+    type: ADD,
+    payload: book,
+  };
+}
+
+export function removeBook(id) {
+  return {
+    type: REMOVE,
+    payload: id,
+  };
+}
+
+export default function reducer(state = INITIAL_STATE, action) {
+  switch (action.type) {
+    case ADD:
+      return [
+        ...state,
+        {
+          id: 1,
+          title: action.payload.title,
+          author: action.payload.author,
+        },
+      ];
+    case REMOVE:
+      return state.filter((book) => book.id !== action.payload.id);
```

Comment:
- [ ] For the **`ADD`** type, instead of hardcoding the `id` (every book id will be `1`) and having this long implementation, we can refactor it to be more concise to use the `action.payload`, like so:

  ```
  case ADD:
    return [
      ...state,
      action.payload,
    ];
  ```

- [ ] The **`REMOVE`** type can also be refactored. In your `removeBook` action creator function the book id has already been passed and reassigned as `payload`, that means `action.payload.id` will return error in the future. I would recommend you use only the `action.payload` like so:
  
  ```
  case REMOVE:
    return state.filter((book) => book.id !== action.payload);
  ```

Reply:
Sure thanks for the hints